mitmproxy/io/compat.py

```diff
@@ -174,19 +174,22 @@ def convert_6_7(data):
 
 def convert_7_8(data):
     data["version"] = 8
-    data["request"]["trailers"] = None
-    if data["response"] is not None:
+    if "request" in data and data["request"] is not None:
+        data["request"]["trailers"] = None
+    if "response" in data and data["response"] is not None:
         data["response"]["trailers"] = None
     return data
 
 
 def convert_8_9(data):
     data["version"] = 9
-    data["request"].pop("first_line_format")
-    data["request"]["authority"] = b""
-    is_request_replay = data["request"].pop("is_replay", False)
+    is_request_replay = False
+    if "request" in data:
```

Comment:
should this also have the check for "not None" for consistency?
(or remove it everywhere? i can't tell if there could ever be an existing key with "None" value).

Reply:
Let's keep it as-is. Eventually we need to tighten down things in a future flow version. :)